deltacat/compute/converter/equality_delete_to_position_delete_session.py

```diff
@@ -0,0 +1,136 @@
+)
+import ray
+import functools
+from deltacat.compute.converter.utils.convert_task_options import (
+    convert_resource_options_provider,
+)
+import logging
+from deltacat import logs
+from collections import defaultdict
+from deltacat.compute.converter.model.convert_session_params import ConvertSessionParams
+from deltacat.compute.converter.constants import DEFAULT_MAX_PARALLEL_DATA_FILE_DOWNLOAD
+from deltacat.compute.converter.steps.convert import convert
+from deltacat.compute.converter.model.convert_input import ConvertInput
+from deltacat.compute.converter.pyiceberg.overrides import (
+    fetch_all_bucket_files,
+    parquet_files_dict_to_iceberg_data_files,
+)
+from deltacat.compute.converter.utils.converter_session_utils import (
+    check_data_files_sequence_number,
+)
+from deltacat.compute.converter.pyiceberg.replace_snapshot import (
+    commit_overwrite_snapshot,
+)
+from deltacat.compute.converter.pyiceberg.catalog import load_table
+
+logger = logs.configure_deltacat_logger(logging.getLogger(__name__))
+
+
+def convert_equality_deletes_to_position_deletes(
+    params: ConvertSessionParams, **kwargs
+):
+    """
+    Convert equality delete to position delete.
+    Compute and memory heavy work from downloading equality delete table and compute position deletes
+    will be executed on Ray remote tasks.
+    """
+
+    catalog = params.catalog
+    table_name = params.iceberg_table_name
+    iceberg_table = load_table(catalog, table_name)
+    data_file_dict, equality_delete_dict, pos_delete_dict = fetch_all_bucket_files(
+        iceberg_table
+    )
+
+    # files_for_each_bucket: {partition_value: [(equality_delete_files_list, data_files_list, pos_delete_files_list)]
+    files_for_each_bucket = defaultdict(tuple)
+    for k, v in data_file_dict.items():
+        logger.info(f"data_file: k, v:{k, v}")
+    for k, v in equality_delete_dict.items():
+        logger.info(f"equality_delete_file: k, v:{k, v}")
+    for partition_value, equality_delete_file_list in equality_delete_dict.items():
+        (
+            result_equality_delete_file,
+            result_data_file,
+        ) = check_data_files_sequence_number(
+            data_files_list=data_file_dict[partition_value],
+            equality_delete_files_list=equality_delete_dict[partition_value],
+        )
+        logger.info(f"result_data_file:{result_data_file}")
+        logger.info(f"result_equality_delete_file:{result_equality_delete_file}")
+        files_for_each_bucket[partition_value] = (
+            result_data_file,
+            result_equality_delete_file,
+            [],
+        )
+
+    iceberg_warehouse_bucket_name = params.iceberg_warehouse_bucket_name
+    print(f"iceberg_warehouse_bucket_name:{iceberg_warehouse_bucket_name}")
+    merge_keys = params.merge_keys
+    # Using table identifier fields as merge keys if merge keys not provided
+    if not merge_keys:
+        # identifier_fields = iceberg_table.schema().identifier_field_names()
```

Comment:
Can we remove commented code? A useful comment is fine but no commented code please. Applies everywhere.